### PR TITLE
Strings section

### PR DESCRIPTION
**Pull request by CGMossa**
* I've anochered the user-guide on main via `.Rprofile`.
* Removed `use-try-from`.
* Removed some things that came from _fork_ and isn't on extendr `master` yet.
* Added a small intro to strings in extendr.

### CODE REVIEW

File: conversion.qmd [hunk 1]
```diff
@@ -156,4 +156,29 @@ A slice is a representation of a sequence of elements that are part of a larger
 
 The R runtime performs [string interning](https://en.wikipedia.org/wiki/String_interning) to
 all of its string elements. This means, that whenever R encounters a new string,
-it adds it to its internal string intern pool. 
\ No newline at end of file
+it adds it to its internal string intern pool. Let's look at a concrete example:
```
Comment (JosiahParry) [inline conversion.qmd]: what is a "string intern pool"

Can we add a callout-tip here?

Comment (CGMossa) [reply conversion.qmd]: Done

**Comment by CGMossa** [comment]
Alright, the purpose of this section is to explain _why_ we cannot provide `&[&str]` as well as why we cannot provide `&mut str`.

**Comment by CGMossa** [comment]
> Alright, the purpose of this section is to explain _why_ we cannot provide `&[&str]` as well as why we cannot provide `&mut str`.

**Pull request closed without merging**